File: ch1/URLify/python/urlify.py
import sys

def URLify(str):
    tmp = ""
    count = 0
    for i in range (0, len(str)):
        if(str[i] == ' '):
            tmp += str[count : i] + "%20"
            count = i+1
    tmp += str[count : len(str)]
    str = tmp
    print str

def main():
    if(len(sys.argv) < 2):
        print("Usage: $ python urlify.py word1 word2 word3 etc")
    else:
        for arg in sys.argv[1:]:
            URLify(arg)

if __name__ == "__main__":
    main()

File: ch1/URLify/java/URLify.java
public class URLify{
    public static String URLify(String url){
        String tmp = "";
        for(int i = 0; i < url.length(); i++){
            if(url.charAt(i) == ' '){
                tmp += "%20";
            } else {
                tmp += url.charAt(i);
            }
        }
        return tmp;
    }

    public static void main(String[] args){
        String a = "Mr John Smith";
        System.out.println(URLify(a));
    }
}
